connection.go

```diff
@@ -3,43 +3,52 @@ package fargo
 // MIT Licensed (see README.md) - Copyright (c) 2013 Hudl <@Hudl>
 
 import (
-	"math/rand"
 	"sync"
 	"time"
 )
 
-func init() {
-	rand.Seed(time.Now().UnixNano())
-}
-
-// SelectServiceURL gets a eureka instance based on the connection's load
-// balancing scheme.
-// TODO: Make this not just pick a random one.
+// SelectServiceURL gets a eureka instance using a round robin algorithm
 func (e *EurekaConnection) SelectServiceURL() string {
+
 	if e.discoveryTtl == nil {
 		e.discoveryTtl = make(chan struct{}, 1)
 	}
-	if e.DNSDiscovery && len(e.discoveryTtl) == 0 {
-		servers, ttl, err := discoverDNS(e.DiscoveryZone, e.ServicePort, e.ServerURLBase)
-		if err != nil {
-			return choice(e.ServiceUrls)
+
+	if e.shouldUseDNSDiscovery() {
+		if err := e.fillServiceURLSWithDNSDiscovery(); err != nil {
+			log.Fatal(err)
```

Comment:
Should there be a return statement in the case of an error? If the ttl has passed, would continuing with the logic below mean we would be returning a stale url? Or is returning the url still worth-while because it might still work?

Reply:
log.fatal will stop the execution flow. Returning an error is superfluous.